src/workers/parquet-exporter.ts

```diff
@@ -15,92 +15,157 @@
+          coreDbPath: this.coreDbPath,
+        },
+      });
 
-      const duckDbSchema = readFileSync(
-        './src/database/duckdb/schema.sql',
-        'utf8',
-      );
-      await db.exec(duckDbSchema);
+      let startTime: number;
 
-      logger.debug('DuckDB schema created');
-    } catch (error) {
-      logger.error('Error creating DuckDB database', error);
-      throw error;
-    }
+      this.worker.on('online', () => {
+        startTime = Date.now();
 
-    try {
-      await db.exec(`INSTALL sqlite; LOAD sqlite;`);
-      await db.exec(
-        `ATTACH '${sqliteBundlesDbPath}' AS sqlite_bundles_db (TYPE sqlite);`,
-      );
-      await db.exec(
-        `ATTACH '${sqliteCoreDbPath}' AS sqlite_core_db (TYPE sqlite);`,
-      );
+        this.log.info('Started Parquet export', {
+          outputDir,
+          startHeight,
+          endHeight,
+          maxFileRows,
+        });
 
-      logger.debug('SQLite databases loaded');
-    } catch (error) {
-      logger.error('Error loading SQLite databases', error);
-      throw error;
-    }
+        this.worker?.postMessage('start');
+      });
 
-    logger.info('DuckDB database created!');
+      this.worker.on('message', (message: any) => {
+        if (message.eventName === EXPORT_COMPLETE) {
+          const endTime = Date.now();
+          const duration = (endTime - startTime) / 1000; // Convert to seconds
 
-    return new ParquetExporter({
-      log: logger,
-      db,
-      duckDbPath,
+          this.log.info(
+            `Parquet export completed in ${duration.toFixed(2)} seconds`,
+            { outputDir, startHeight, endHeight, maxFileRows },
+          );
+
+          this.isExporting = false;
+          this.worker?.terminate();
+          resolve();
+        } else if (message.eventName === EXPORT_ERROR) {
+          this.isExporting = false;
+          this.worker?.terminate();
+          this.log.error('Parquet export error', {
+            error: message.error,
+            stack: message.stack,
+          });
+          reject(new Error(message.error));
+        }
+      });
+
+      this.worker.on('error', (error) => {
+        this.log.error('Worker error', error);
+        this.isExporting = false;
+        reject(error);
+      });
+
+      this.worker.on('exit', (code) => {
+        if (code !== 0) {
+          this.isExporting = false;
+          reject(new Error(`Worker stopped with exit code ${code}`));
+        }
+      });
     });
   }
 
-  private async importBlocks({
-    startHeight,
-    endHeight,
-  }: {
-    startHeight: number;
-    endHeight: number;
-  }) {
-    const log = this.log.child({ method: 'importBlocks' });
-    const query = `
+  async stop(): Promise<void> {
+    const worker = this.worker;
+
+    if (worker) {
+      return new Promise((resolve) => {
+        worker.on('exit', () => {
+          resolve();
+        });
+        worker.postMessage('terminate');
+      });
```

Comment:
@karlprieb Why are we using the `terminate` message here? I'm fairly certain we're doing this elsewhere too, but I wonder if it's the right approach.

Reply:
I'm not sure why we decided to go with `terminate` message and `process.exit(0)` instead of just calling `worker.terminate()`.

Checking `terminate` in nodejs docs (https://nodejs.org/api/worker_threads.html#workerterminate):
> Stop all JavaScript execution in the worker thread as soon as possible. Returns a Promise for the exit code that is fulfilled when the ['exit' event](https://nodejs.org/api/worker_threads.html#event-exit) is **emitted.**

Testing here it seems to have the same behavior as calling `process.exit(0)` manually. In that case I rather use `worker.terminate()` as it's part of the workers api and can be optimized in the future (not sure).

Reply:
[`09b81c2` (#215)](https://github.com/ar-io/ar-io-node/pull/215/commits/09b81c2b03ab84040fe48b652f7d1efb8714a964)